packages/currency/src/Balance.ts

```diff
@@ -69,19 +77,29 @@ export default class Balance<T extends BaseCurrencyType> {
     const groupSeparator = options?.groupSeparator || ','
     const showTicker = options?.showTicker === undefined ? true : options.showTicker
     const format = { decimalSeparator, groupSeparator, groupSize: 3 }
+    const roundingMode = options?.roundingMode || BigNumber.ROUND_DOWN
+    const decimalPlacesToDisplay = getVal(maxDecimalPlaces, this.type.format?.decimalPlaces)
+    const keepTrailingZeroes = getVal(
+      options?.showTrailingZeroes,
+      this.type.format?.showTrailingZeroes,
+    )
 
     let numberString = ''
-    if (maxDecimalPlaces !== undefined && maxDecimalPlaces !== null) {
-      numberString = this.bigBalance.toFormat(
-        maxDecimalPlaces,
-        options?.roundingMode || BigNumber.ROUND_DOWN,
-        format,
-      )
+    if (decimalPlacesToDisplay !== undefined && decimalPlacesToDisplay !== null) {
```

Comment:
```suggestion
    if (decimalPlacesToDisplay) {
```

Reply:
This doesn't work because I need it to go into that if statement if `decimalPlacesToDisplay === 0`